compiler/damlc/daml-doc/test/DA/Daml/GHC/Damldoc/Tests.hs

```diff
@@ -280,35 +282,12 @@ fileTest damlFile = do
 
   if null expectations
     then pure []
-    else do input <- T.readFileUtf8 damlFile
-            mapM (goldenTest input) expectations
-
-  where goldenTest :: T.Text -> FilePath -> IO Tasty.TestTree
-        goldenTest input expectation =
-          let check docs = do
-                let extension = takeExtension expectation
-                ref <- T.readFileUtf8 expectation
-                case extension of
-                  ".rst"  -> expectEqual extension ref $ renderPage renderRst $ renderModule docs
-                  ".md"   -> expectEqual extension ref $ renderPage renderMd $ renderModule docs
-                  ".json" -> expectEqual extension ref
-                             (T.decodeUtf8 . BS.toStrict $
-                               AP.encodePretty' jsonConf docs)
-                  other  -> error $ "Unsupported file extension " <> other
-
-              expectEqual :: String -> T.Text -> T.Text -> Assertion
-              expectEqual extension ref actual
-                | standardizeEoL ref == standardizeEoL actual = pure ()
-                | otherwise = do
-                    let actualFile = replaceExtensions expectation ("ACTUAL" <> extension)
-                        asLines = lines . T.unpack
-                        diff = ppDiff $ getGroupedDiff (asLines ref) (asLines actual)
-
-                    T.writeFileUtf8 actualFile actual
-                    assertFailure $
-                      "Unexpected difference between " <> expectation <>
-                      " and actual output.\n" <>
-                      "Unexpected output has been written to " <> actualFile <> "\n" <>
-                      "Differences:\n" <> diff
-
-          in pure $  damldocExpect (Just $ takeDirectory damlFile) ("File: " <> expectation) [input] check
+    else do
+      doc <- runDamldoc damlFile (Just $ takeDirectory damlFile)
+      pure $ flip map expectations $ \expectation ->
+        goldenVsString ("File: " <> expectation) expectation $ pure $
```

Comment:
Does the diff in the case of a failure look reasonable? IIRC this was rather confusing at some point when the difference was only whitespace. We “fixed” that by throwing a bunch of `show` at it but it would be nice to at least not regress here.

Reply:
I changed it to use `goldenVsStringDiff`. Now the output looks like
```
      1c1
      < # <a name="module-ioudsl-47944"></a>Module IouDSL
      ---
      > # <a name="module-proposaliou-96142"></a>Module ProposalIou
      5c5
      < <a name="type-ioudsl-iou-73876"></a>**template** [Iou](#type-ioudsl-iou-73876)
      ---
      > <a name="type-proposaliou-iou-51326"></a>**template** [Iou](#type-proposaliou-iou-51326)
      23c23
      < <a name="type-ioudsl-proposaliou-92778"></a>**template instance** [ProposalIou](#type-ioudsl-proposaliou-92778)
      ---
      > <a name="type-proposaliou-proposaliou-81988"></a>**template instance** [ProposalIou](#type-proposaliou-proposaliou-81988)
      25c25
      < > = Proposal [Iou](#type-ioudsl-iou-73876)
      ---
      > > = Proposal [Iou](#type-proposaliou-iou-51326)
```
I think that's good enough.

Reply:
Thanks!

Reply:
> ```
>       1c1
>       < # <a name="module-ioudsl-47944"></a>Module IouDSL
>       ---
>       > # <a name="module-proposaliou-96142"></a>Module ProposalIou
>       5c5
>       < <a name="type-ioudsl-iou-73876"></a>**template** [Iou](#type-ioudsl-iou-73876)
>       ---
>       > <a name="type-proposaliou-iou-51326"></a>**template** [Iou](#type-proposaliou-iou-51326)
>       23c23
>       < <a name="type-ioudsl-proposaliou-92778"></a>**template instance** [ProposalIou](#type-ioudsl-proposaliou-92778)
>       ---
>       > <a name="type-proposaliou-proposaliou-81988"></a>**template instance** [ProposalIou](#type-proposaliou-proposaliou-81988)
>       25c25
>       < > = Proposal [Iou](#type-ioudsl-iou-73876)
>       ---
>       > > = Proposal [Iou](#type-proposaliou-iou-51326)
> ```

😍